src/nvim/testdir/test_escaped_glob.vim

```diff
@@ -0,0 +1,32 @@
+" Test whether glob()/globpath() return correct results with certain escaped
+" characters.
+
+function SetUp()
+  " make sure glob() doesn't use the shell
+  set shell=doesnotexist
+  " consistent sorting of file names
+  set nofileignorecase
+endfunction
+
+function Test_glob()
+  if !has('unix')
+    " This test fails on Windows because of the special characters in the
+    " filenames. Disable the test on non-Unix systems for now.
+    return
+  endif
+  call assert_equal("", glob('Xxx\{'))
+  call assert_equal("", glob('Xxx\$'))
+  w! Xxx{
+  w! Xxx\$
+  call assert_equal("Xxx{", glob('Xxx\{'))
+  call assert_equal("Xxx$", glob('Xxx\$'))
+  call delete('Xxx{')
+  call delete('Xxx$')
+endfunction
+
+function Test_globpath()
```

Comment:
Fails on Appveyor because `globpath` outputs paths with backslash. This passes for Vim because of `set shellslash` but that breaks `shellescape` for cmd.exe. Neovim uses cmd.exe, not sh, for its shell on Windows.

Reply:
But in `SetUp` it does `set shell=doesnotexist` to not use a shell?

Will try the following now:
```diff
modified: src/nvim/testdir/test_escaped_glob.vim
─────────────────────────────────────────────────────────────────────────────────────────────
@ src/nvim/testdir/test_escaped_glob.vim:28 @ function Test_glob()
 endfunction
 
 function Test_globpath()
-  call assert_equal("sautest/autoload/footest.vim",
+  let slash = (!exists('+shellslash') || &shellslash) ? '/' : '\'
+  call assert_equal('sautest'.slash.'autoload'.slash.'footest.vim',
         \ globpath('sautest/autoload', '*.vim'))
-  call assert_equal(['sautest/autoload/footest.vim'],
+  call assert_equal(['sautest'.slash.'autoload'.slash.'footest.vim'],
         \ globpath('sautest/autoload', '*.vim', 0, 1))
 endfunction
```

Reply:
I use `expand` myself to make it less messy when merging patches but that should be fine.

Reply:
@blueyed `set shell` has no effect on the value of `shellslash`.

Reply:
What should be done here then now?

You said:
> This passes for Vim because of set shellslash

Where does Vim use `set shellslash`?  (at least not in the patch/test directly - https://github.com/vim/vim/commit/292eff0c5aacb8531d65509679b6c29eae8dc22a#diff-fa96be7b9f8aeee55448425e4aa092f2)

Do you mean that Vim uses `set shell`?